### ISSUE TITLE
Support for "azurerm_storage_share" data source

### ISSUE THREAD

**Issue by ccontino84** [terraform-providers/terraform-provider-azurerm#4931]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When I try to declare a data source for a azurerm_storage_share resource I get the following error:
> The provider provider.azurerm does not support data source
"azurerm_storage_share".

### New or Affected Resource(s)

* azurerm_storage_share

### Potential Terraform Configuration

```hcl
data "azurerm_storage_share" "storage_share" {
  name                = "storage-share"
  storage_account_name = "${var.storage_account_name}
}
```

### References

* #0000

**Comment by r0bnet** [comment]
Hey @ccontino84,

looking at the docs for the storage_share resource it says that `resource_group_name` is deprecated and the property `storage_account_name` is the preferred approach. Would you mind to change that in favor of the resource conventions?

**Comment by ccontino84** [comment]
Thanks @r0bnet, it was a copy paste mistake. Now I updated the description according to docs.

**Comment by r0bnet** [comment]
@ccontino84 great. This looks valid for me. Can you pls add which attributes you'd like to have exposed?
Per default i'd propose to expose `id` and `url`. Is that enough for you?

**Comment by ccontino84** [comment]
@r0bnet for my use case the ```name``` is already enough, because that's all I need to configure a ```azurerm_storage_share_directory```, but I guess also ```id``` and ```url``` could be useful in a different scenario.

**Comment by r0bnet** [comment]
@ccontino84 but if you simply want the name AND you need the name for the data source then i don't see the problem? To create a `share_directory` you could easily pass the name of the share and the name of the storage account directly?

**Comment by ccontino84** [comment]
@r0bnet yes, but in our configuration the ```share_directory``` is created in a different terraform project (it's a quite complicated architecture I know)

**Comment by a8ree** [comment]
I've also got a similar issue where a ML workspace is being created and thus creating a share in a storage account. In order that I can setup the file share backup, I need to know the name of the share.

ML workspace creates a share with a random guid
e.g.
azureml-filestore-60119c34-fd89-4a57-bb0c-2adbe07d682a

**Comment by infa-pavakuma** [comment]
Any Update on this ?

**Comment by katbyte** [comment]
closed by #12693
